superset-frontend/src/explore/components/SaveModal.tsx

```diff
@@ -270,6 +327,22 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
               data-test="new-chart-name"
             />
           </FormItem>
+          {this.props.datasource && this.props.datasource.type === 'query' ? (
+            <FormItem label={t('Dataset Name')} required>
+              <InfoTooltipWithTrigger
+                tooltip={t('A reusable dataset will be saved with your chart.')}
+                placement="right"
+              />
+              <Input
+                name="dataset_name"
+                type="text"
+                placeholder="Dataset Name"
+                value={this.state.datasetName}
+                onChange={this.handleDatasetNameChange}
+                data-test="new-dataset-name"
+              />{' '}
```

Comment:
What is the use case for the empty whitespace `{' '}` here?  If using to add padding, using emotion to add padding or margin to right of the input will make this more controlled

Reply:
same as above comment